cmake/AwsSanitizers.cmake

```diff
@@ -32,23 +42,37 @@ function(aws_check_sanitizer sanitizer)
 
         # Need to set this here so that the flag is passed to the linker
         set(CMAKE_REQUIRED_FLAGS ${sanitizer_test_flag})
-        check_c_compiler_flag(${sanitizer_test_flag} ${out_variable})
+        if(ARG_IS_CXX)
+            check_cxx_compiler_flag(${sanitizer_test_flag} ${out_variable})
+        else()
+            check_c_compiler_flag(${sanitizer_test_flag} ${out_variable})
+        endif()
     else()
         set(${out_variable} 0 PARENT_SCOPE)
     endif()
 endfunction()
 
 # This function enables sanitizers on the given target
-# Options:
+# Parameters:
 #  SANITIZERS: The list of extra sanitizers to enable
+# Options:
+#  IS_CXX: This option switches the function to check the CXX compiler.
```

Comment:
cant you just check whether the target is getting compiled for c or c++?
something like ```get_target_property(target_lang mytarget LANGUAGE)```

Reply:
Hmmm, it seems there is no universal property for a target language (which is kinda unexpected). There is the `LINKER_LANGUAGE` property, but it's optional.

Also, the `aws_check_sanitizer` function doesn't have `target` as a parameter, so we can't use this approach there.

I also can't find a way to get a cmake project language.
So, the only alternative approach I can think of is to check the enabled languages, and use `check_cxx_compiler_flag` if C++ is present. But I'm not sure it won't cause some issues in in the projects where both C and C++ are used.

Reply:
maybe try CMAKE_CXX_COMPILER_LOADED and CMAKE_C_COMPILER_LOADED. seems to work fine on my machine and as far as i can tell should be cross platform

Reply:
Yep, `CMAKE_CXX_COMPILER_LOADED` is [available since at least CMake 3.0](https://github.com/Kitware/CMake/blob/v3.0.0/Modules/CMakeCXXCompiler.cmake.in#L13) and I don't see any platform specifics there.

One thing though. It is a global variable, so when a C++ project uses a C project via `add_subdirectory`, `CMAKE_CXX_COMPILER_LOADED` will be defined for the C project. For example, `aws-c-common` used by `aws-iot-device-sdk-cpp-v2` sees C++ enabled:

```
$ cmake -S aws-iot-device-sdk-cpp-v2/ -B cmake-build-tsan -DENABLE_SANITIZERS=ON -DSANITIZERS=thread
...
- aws-c-common CMakeLists.txt: calling aws_add_sanitizers
- aws_check_sanitizer: CMAKE_CXX_COMPILER_LOADED is defined
- aws_check_sanitizer: CMAKE_C_COMPILER_LOADED is defined
...
```

I think, this shouldn't cause any issues in practice, but I just don't feel comfortable with checking C++ compiler flags in a C project.

That said, if you're strongly against adding `IS_CXX` option, I'll use `CMAKE_CXX_COMPILER_LOADED`. Or, there is also [ENABLED_LANGUAGES](https://cmake.org/cmake/help/latest/prop_gbl/ENABLED_LANGUAGES.html) property, which is also suitable here and it's documented.

Reply:
Thinking more about this, checking C++ compiler in C subprojects can't cause any issues, because C++ compiler should be checked anyway. So yeah, I'll just add a check for CMAKE_CXX_COMPILER_LOADED.